revscoring/scoring/statistics/classification/scaled_prediction_statistics.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from collections import OrderedDict
+
+from tabulate import tabulate
+
+from ... import util
+from .scaled_classification_matrix import ScaledClassificationMatrix
+
+logger = logging.getLogger(__name__)
+
+
+class ScaledPredictionStatistics(ScaledClassificationMatrix):
+    FIELDS = ['match_rate', 'filter_rate',
+              'precision', '!precision',
+              'recall', '!recall',
+              'accuracy', 'fpr',
+              'f1', '!f1']
+
+    def format_json(self, path_tree, ndigits=3, **kwargs):
+        return OrderedDict(
+            (field, util.round(self[field], ndigits))
+            for field in (path_tree.keys() or self.keys()))
+
+    def format_str(self, path_tree, ndigits=3, **kwargs):
+        table_data = [[util.round(self[field], ndigits)
+                       for field in path_tree.keys() or self.keys()]]
+        return tabulate(table_data, headers=path_tree.keys() or self.keys())
+
+    def __getitem__(self, field):
+        if field in self.FIELDS:
+            method_name = field.replace("!", "_")
+            return getattr(self, method_name)()
+        else:
+            raise KeyError(field)
+
+    def keys(self):
+        return self.FIELDS
+
+    def __iter__(self):
+        return iter(self.keys())
+
+    def match_rate(self):
+        """
+        The proportion of observations that are matched in prediction.
+
+            match-rate = positives / n
+        """
+        return (self.positives / self.n) if self.n is not 0 else None
+
+    def filter_rate(self):
+        """
+        The proportion of observations that are not matched.
+
+            filter-rate = 1 - match-rate
+        """
+        return (1 - self.match_rate()) \
+               if self.match_rate() is not None else None
```

Comment:
That would imply we are filtering 100% of 0 observations :S  I think it's better to return a non-value than to pretend like that makes sense.